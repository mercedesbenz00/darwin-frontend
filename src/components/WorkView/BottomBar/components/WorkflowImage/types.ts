import { DatasetItemPayload } from '@/store/types'

export type WorkflowImageProps = {
  datasetItem: DatasetItemPayload
}
