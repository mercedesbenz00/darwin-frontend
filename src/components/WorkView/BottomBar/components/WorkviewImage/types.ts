export type WorkviewImageProps = {
  filename: string,
  thumbnail: string,
  selected?: boolean,
  isVideo?: boolean
}
