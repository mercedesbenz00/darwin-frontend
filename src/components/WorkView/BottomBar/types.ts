import { DatasetItemPayload, V2DatasetItemPayload } from '@/store/types'

/**
 * A single item being rendered in the scroller
 *
 * It represents either an image, or a page request for a set of images to be
 * loaded from the backend.
 *
 * The id must be computed in a way that makes the overall collection sortable by it.
 */
export type BottomBarItem = {
  id: string
  data?: DatasetItemPayload
  loading?: boolean,
  size?: number
}

export type BottomBarItemV2 = {
  id: string
  data?: V2DatasetItemPayload
  loading?: boolean,
  size?: number
}
