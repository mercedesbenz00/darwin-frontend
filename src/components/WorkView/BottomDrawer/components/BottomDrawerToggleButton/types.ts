export type BottomDrawerToggleButtonProps = {
  name: string,
  collapsed?: boolean,
  hover?: boolean
}
