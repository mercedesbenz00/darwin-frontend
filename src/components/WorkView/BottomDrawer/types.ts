export type BottomDrawerProps = {
  name: string,
  resizable?: boolean,
  initialHeight: number,
  minHeight: number,
  maxHeight: number,
}
