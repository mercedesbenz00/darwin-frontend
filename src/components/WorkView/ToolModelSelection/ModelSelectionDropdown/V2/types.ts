import { ModelType } from '@/store/types'
import { RGBA } from '@/utils'

export type ModelSelectionDropdownClassOption = {
  color: RGBA
  label: string
}

export type ModelSelectionDropdownOption = {
  id: string
  label: string
  type: ModelType
  classes: ModelSelectionDropdownClassOption[]
}
