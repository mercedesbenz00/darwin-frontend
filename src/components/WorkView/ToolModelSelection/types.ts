import { RGBA } from '@/utils'

export type ClassDescriptor = { label: string, color?: RGBA }
