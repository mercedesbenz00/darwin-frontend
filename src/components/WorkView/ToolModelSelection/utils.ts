import Vue from 'vue'

const extendComponent = (component: any, propsData: any) => {
  const ExtendedComponent = Vue.extend(component)
  return new ExtendedComponent({ propsData })
}

export const mountedComponentHtml = (component: any, propsData: any) => {
  return extendComponent(component, propsData).$mount().$el.innerHTML
}
