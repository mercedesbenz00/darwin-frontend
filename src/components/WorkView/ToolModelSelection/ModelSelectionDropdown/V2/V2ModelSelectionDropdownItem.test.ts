import { createLocalVue, shallowMount, Wrapper } from '@vue/test-utils'
import VTooltip from 'v-tooltip'
import Vue from 'vue'

import { ModelType } from '@/store/types'
import { RGBA } from '@/utils'

import {
  V2ModelSelectionDropdownItem,
  ModelSelectionDropdownOption,
  ModelSelectionDropdownClassOption
} from '.'

const localVue = createLocalVue()
localVue.use(VTooltip, { defaultHtml: true })

type ItemProp = {
  option: ModelSelectionDropdownOption,
  selected?: boolean,
}

let propsData: ItemProp

const classes: ModelSelectionDropdownClassOption[] = [
  { color: { r: 10, g: 10, b: 10, a: 0.1 } as RGBA, label: 'bar' },
  { color: { r: 20, g: 20, b: 20, a: 0.2 } as RGBA, label: 'baz' }
]

const defaultItem: ModelSelectionDropdownOption = {
  id: 'foo-1234',
  label: 'foo',
  type: ModelType.AutoAnnotation,
  classes
}

const itMatchesSnapshot = (propsData: ItemProp): void => it('matches snapshot', () => {
  const wrapper = shallowMount(V2ModelSelectionDropdownItem, { localVue, propsData })
  expect(wrapper).toMatchSnapshot()
})

itMatchesSnapshot({
  option: defaultItem
})

it('is instanciated', () => {
  const _propsData = {
    option: defaultItem
  }

  const wrapper = shallowMount(V2ModelSelectionDropdownItem, { localVue, propsData: _propsData })
  expect(wrapper).toBeTruthy()
})

describe('has correct values', () => {
  let wrapper: Wrapper<Vue>

  const _propsData = {
    option: defaultItem
  }

  beforeEach(() => {
    propsData = _propsData
    wrapper = shallowMount(V2ModelSelectionDropdownItem, { localVue, propsData })
  })

  afterEach(() => {
    wrapper.destroy()
  })

  itMatchesSnapshot(_propsData)

  it('renders correctly', () => {
    expect(wrapper.find('.model__dropdown__item').exists()).toBe(true)
  })

  it('contain the right icon', () => {
    const icon = wrapper.find('.model__dropdown__item__icon')
    expect(icon.exists()).toBe(true)
    expect(icon.attributes().modeltype).toBe('auto_annotate')
  })

  it('contain the right label', () => {
    const label = wrapper.find('.model__dropdown__item__label')
    expect(label.exists()).toBe(true)
    expect(label.text()).toBe('foo')
  })

  it('contain the right classes', () => {
    const badges = wrapper.find('badges-stub')
    expect(badges.exists()).toBe(true)
  })

  it('contain the right status', () => {
    const status = wrapper.find('.model__dropdown__item__status')
    expect(status.attributes().value).toBe('active')
  })
})
