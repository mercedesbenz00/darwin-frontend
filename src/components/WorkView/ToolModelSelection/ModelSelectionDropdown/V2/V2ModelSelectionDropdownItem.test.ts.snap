// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has correct values matches snapshot 1`] = `
<div tabindex="1" autofocus="autofocus" class="model__dropdown__item">
  <div class="model__dropdown__item__col">
    <v2-model-icon-stub modeltype="auto_annotate" class="model__dropdown__item__icon"></v2-model-icon-stub>
  </div>
  <div class="model__dropdown__item__col"><span class="model__dropdown__item__label has-tooltip" data-original-title="null">
      foo
    </span> <span class="model__dropdown__item__classes"><badges-stub items="[object Object],[object Object]"></badges-stub></span></div>
  <div class="model__dropdown__item__col">
    <status-stub value="active" class="model__dropdown__item__status"></status-stub>
  </div>
</div>
`;

exports[`matches snapshot 1`] = `
<div tabindex="1" autofocus="autofocus" class="model__dropdown__item">
  <div class="model__dropdown__item__col">
    <v2-model-icon-stub modeltype="auto_annotate" class="model__dropdown__item__icon"></v2-model-icon-stub>
  </div>
  <div class="model__dropdown__item__col"><span class="model__dropdown__item__label has-tooltip" data-original-title="null">
      foo
    </span> <span class="model__dropdown__item__classes"><badges-stub items="[object Object],[object Object]"></badges-stub></span></div>
  <div class="model__dropdown__item__col">
    <status-stub value="active" class="model__dropdown__item__status"></status-stub>
  </div>
</div>
`;
